### PR TITLE
Update TranslationAveraging to return Pose(3), not Point(3)

### PR DESCRIPTION
**Pull request by johnwlambert**
- Aligns code with the process graph, so that computation occurs in blue boxes, and not in free functions.
- Combines estimated global R and global t in TA, instead of DA, since DA should only create tracks.
- Make MFAS optional.

Pulls in many of the changes from the large https://github.com/borglab/gtsfm/pull/515 PR.

### CODE REVIEW

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 1]
```diff
@@ -137,23 +152,23 @@ def compute_inlier_mask(
...
             if outlier_weights_sum[(i2, i1)] / len(projection_directions) < OUTLIER_WEIGHT_THRESHOLD:
-                inlier_idxs.add((i1, i2))
+                inliers.add((i1, i2))
 
-        return inlier_idxs
+        return inliers
 
     def _get_prior_measurements_in_world_frame(
         self,
         relative_pose_priors: Dict[Tuple[int, int], PosePrior],
```
Comment (ayushbaid) [inline gtsfm/averaging/translation/averaging_1dsfm.py:161]: Can we rename it to `i1Ti2_priors`?

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 2]
```diff
@@ -199,15 +214,15 @@ def run(
...
+    ) -> Tuple[List[Optional[Pose3]], Optional[GtsfmMetricsGroup]]:
         """Run the translation averaging.
 
         Args:
             num_images: number of camera poses.
             i2Ui1_dict: relative unit-translation as dictionary (i1, i2): i2Ui1
             wRi_list: global rotations for each camera pose in the world coordinates.
             absolute_pose_priors: priors on the camera poses (not delayed).
-            relative_pose_priors: priors on the pose between camera pairs (not delayed)
+            relative_pose_priors: priors on the pose between camera pairs (not delayed) as (i1, i2): i1Ti2.
```
Comment (ayushbaid) [inline gtsfm/averaging/translation/averaging_1dsfm.py:225]: Rename to `i1Ti2_priors`

**Review by ayushbaid — COMMENTED** [review]
LGTM overall. We should add a test with prior values.

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 3]
```diff
@@ -168,7 +183,7 @@ def get_prior_in_world_frame(i2, i2Ti1_prior):
             # noise_model = gtsam.noiseModel.Gaussian.Covariance(i2Ti1_prior.covariance)
             noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)
             w_i2ti1_priors.append(
-                gtsam.BinaryMeasurementPoint3(
+                BinaryMeasurementPoint3(
                     i2,
                     i1,
                     get_prior_in_world_frame(i2, i2Ti1_prior),
```
Comment (johnwlambert) [inline gtsfm/averaging/translation/averaging_1dsfm.py:189]: @akshay-krishnan @ayushbaid we refer to these as i2Ti1 priors here, but I thought they were i1Ti2 priors?

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 4]
```diff
@@ -61,18 +72,21 @@ def __init__(
         self,
         robust_measurement_noise: bool = True,
         projection_sampling_method: ProjectionSamplingMethod = ProjectionSamplingMethod.SAMPLE_WITH_UNIFORM_DENSITY,
+        MFAS_outlier_rejection: bool = True,
```
[the remark this answers is not in the document]
Comment (akshay-krishnan) [reply gtsfm/averaging/translation/averaging_1dsfm.py:75]: also this should probably come before projection_sampling_method, and projection_sampling_method would be irrelevant if MFAS_outlier_rejection=False.

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 5]
```diff
@@ -137,38 +152,38 @@ def compute_inlier_mask(
...
+        if len(i1Ti2_priors) == 0:
             return gtsam.BinaryMeasurementsPoint3()
 
         def get_prior_in_world_frame(i2, i2Ti1_prior):
             return wRi_list[i2].rotate(i2Ti1_prior.value.translation())
 
         w_i2ti1_priors = gtsam.BinaryMeasurementsPoint3()
 
-        for (i1, i2), i2Ti1_prior in relative_pose_priors.items():
+        for (i1, i2), i2Ti1_prior in i1Ti2_priors.items():
```
Comment (akshay-krishnan) [inline gtsfm/averaging/translation/averaging_1dsfm.py:181]: now that is a confusing line..

**Review by akshay-krishnan — APPROVED** [review]
This almost looks good John, can we revive this PR and submit it, so we can work on other changes to this file?

File: gtsfm/averaging/translation/averaging_1dsfm.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/averaging/translation/averaging_1dsfm.py]: Now renamed to `i2Ti1_priors`, as it looks like we are pulling out `i2Ti1` from the dictionary.

File: gtsfm/averaging/translation/averaging_1dsfm.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/averaging/translation/averaging_1dsfm.py]: Now renamed to `i2Ti1_priors`, as it looks like we are pulling out `i2Ti1` from the dictionary.

**Pull request merged**